assertj-core/src/test/java/org/assertj/core/api/string_/StringAssert_asByte_Test.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api.string_;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import org.assertj.core.api.StringAssert;
+import org.assertj.core.api.StringAssertBaseTest;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link StringAssert#asByte()}</code> and <code>{@link StringAssert#asByte(int)}</code>.
+ *
+ * @author HeZean
+ */
+class StringAssert_asByte_Test extends StringAssertBaseTest {
+
+  @Override
+  protected StringAssert invoke_api_method() {
+    // Verify disabled as the asByte cast throws an AssertionError when the assertion's string is not a valid byte.
+    // Tested as below.
+    return null;
+  }
+
+  @Override
+  protected void verify_internal_effects() {
+    // Verify disabled as the asByte cast have no internal effect.
+  }
+
+  @Override
+  public void should_return_this() {
+    // Test disabled as the assertion does not return this.
+  }
+
+  @Test
+  void should_parse_string_as_byte_when_the_numeric_string_does_not_overflow() {
+    // GIVEN
+    String string = "127";
+    // THEN
+    assertThat(string).asByte().isEqualTo((byte) 127);
+
+    // GIVEN
+    string = "-128";
+    // THEN
+    assertThat(string).asByte().isEqualTo((byte) -128);
+
+    // GIVEN
+    string = "1001";
+    // THEN
+    assertThat(string).asByte(2).isEqualTo((byte) 9);
+  }
```

Comment:
convert to a parameterized test https://www.baeldung.com/parameterized-tests-junit-5